internal/pkg/cli/pipeline_init.go

```diff
@@ -355,27 +353,56 @@ func relPath(fullPath string) string {
 	return relPath
 }
 
-// TODO: Nice-to-have: have an opts.listRemoteRepos() method that execs out to `git remote -v` and parse repo name to offer select menu
-func (opts *InitPipelineOpts) selectGitHubRepo() error {
-	repo, err := opts.prompt.Get(
-		pipelineEnterGitHubRepoPrompt,
-		fmt.Sprintf(`The GitHub repository linked to your workspace. Pushing to this repository will trigger your pipeline build stage. Please enter full repository URL, e.g. "https://github.com/myCompany/myRepo", or the owner/rep, e.g. "myCompany/myRepo"`),
-		validateGitHubRepo,
+func (opts *InitPipelineOpts) selectGitHubURL() error {
+	url, err := opts.prompt.SelectOne(
+		pipelineSelectGitHubURLPrompt,
+		pipelineSelectGitHubURLHelpPrompt,
+		opts.repoURLs,
 	)
-
-	// Remove ".git" at the end of the Github repo URL.
-	repoNameSplit := strings.Split(repo, ".")
-	if repoNameSplit[len(repoNameSplit)-1] == "git" {
-		repo = strings.Join(repoNameSplit[:len(repoNameSplit)-1], ".")
+	if err != nil {
+		return fmt.Errorf("select GitHub URL: %w", err)
 	}
+	opts.GitHubURL = url
 
-	if err != nil {
-		return fmt.Errorf("failed to get GitHub repository: %w", err)
+	return nil
+}
+
+func (opts *InitPipelineOpts) parseOwnerRepoName(url string) (string, string, error) {
+	parsedURL := strings.TrimPrefix(url, opts.regexZeroLengthAssertion(url, `.*(github.com)(:|\/)`, "", ""))
+	ownerRepo := strings.Split(parsedURL, string(os.PathSeparator))
+	if len(ownerRepo) != 2 {
+		return "", "", fmt.Errorf(`unable to parse the owner and repository for "%s", please pass in your repository URL with -u`, url)
 	}
+	return ownerRepo[0], ownerRepo[1], nil
+}
 
-	opts.GitHubRepo = repo
+func (opts *InitPipelineOpts) regexZeroLengthAssertion(s string, pattern string, prefix string, suffix string) string {
+	regexPattern := regexp.MustCompile(pattern)
+	regexResult := regexPattern.FindString(s)
+	resultWithoutSuffixAndPrefix := strings.TrimSuffix(strings.TrimPrefix(regexResult, prefix), suffix)
+	return resultWithoutSuffixAndPrefix
+}
 
-	return nil
+// examples:
+// efekarakus	git@github.com:efekarakus/grit.git (fetch)
+// efekarakus	https://github.com/karakuse/grit.git (fetch)
+// origin	https://github.com/koke/grit (fetch)
+// koke      git://github.com/koke/grit.git (push)
+func (opts *InitPipelineOpts) parseGitRemoteResult(s string) ([]string, error) {
+	var urls []string
+	urlSet := make(map[string]bool)
+	items := strings.Split(s, "\n")
+	for _, item := range items {
```

Comment:
Should we skip the item if it doesn't contain "github.com"?

Reply:
Good idea!